src/main/bash/sdkman-env.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+#
+#   Copyright 2017 Marco Vermeulen
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+function __sdk_env {
+  if [[ ! -f .sdkmanrc ]]; then
+    __sdkman_echo_red '.sdkmanrc not found.'
+
+    return 1
+  fi
+
+  local candidate version
+
+  while IFS='=' read -r candidate version; do
+    __sdk_use "$candidate" "$version"
+  done < .sdkmanrc
+
+  # if the last line is not terminated by a newline character, `candidate` and
+  # `version` will be set but `read` will return a non-zero status code,
+  # resulting in the loop body not being executed
```

Comment:
Lets try finding a cleaner solution for this if we can. And maybe some tests to cover this would also be good. Perhaps if we take responsible for generating this file with an `sdk env` command extension we wouldn't need to be so defensive.

For now I would suggest failing hard with a proper error message rather than trying to brute-force the last line. wdyt?

Reply:
The cleaner solution would be to not use Bash, but that is out of the question. 
Other languages, like AWK, Groovy, you name it, handle this case just fine.

The idea is taken from http://mywiki.wooledge.org/BashFAQ/001#CA-47120ef7c8534fbb8d04287560707aee3db999b5_1.

Alternatively we could rewrite the expression to

```sh
while IFS='=' read -r candidate version || [[ -n $candidate ]]; do
  __sdk_use "$candidate" "$version"
done < .sdkmanrc
```
Same logic, same flaws, though 😄.

I think the check/hack here is better than throwing an error because people will write/modify the file by hand, even when providing a way to do this with `sdkman`. This prevents the script from breaking in that case.

Also, it seems to be impossible to find out which error led read to fail.

Reply:
Also I was thinking, if we should not also

* strip leading / trailing whitespace
* allow comments

Too much?

Reply:
In any case we should write tests for each case.

Reply:
Should we still handle the valid entries?

Reply:
I would refuse to process the file until corrected. wdyt?

    No newline at the end of the file. The .sdkmanrc will be ignored until corrected.

Or something like this.

Reply:
I am a little torn on this. Most tools handle that case (missing newline) just fine (even older ones like AWK).

My proposed fix (even if quite ugly) would work in those cases and people would be happy. Otherwise you'd may see questions regarding the missing newline pop up on StackOverflow or GitHub (StackOverflow is actually full of questions regarding similar warnings / errors).

Reply:
See https://mywiki.wooledge.org/BashFAQ/001#My_text_files_are_broken.21__They_lack_their_final_newlines.21

which descrives the fix.

Reply:
There do seem to be other ways of getting around the problem. See here:
https://stackoverflow.com/questions/1521462/looping-through-the-content-of-a-file-in-bash

In particular, this approach looks promising:

```
cat peptides.txt | while read line || [[ -n $line ]];
do
   # do something with $line here
done
```

Reply:
Changed it to 

```sh
  while IFS='=' read -r candidate version || [[ -n "$candidate" ]]; do
    __sdk_use "$candidate" "$version"
  done < "$sdkmanrc"
```

which is basically the solution you proposed, excluding the `cat .sdkmanrc` (I use a redirection instead as it performs better than a `cat` followed by a `|`).